### PR TITLE
Adds Error label to inputs

### PR DESCRIPTION
**Pull request by Mera-Gangapersaud**
# Description 

The Moments Page now displays an error message below the input if it's left empty

## More Details

I added an Error Label Component for Inputs so the error message shows up below the textInputArea.

## Corresponding Issue

#1085 

# Screenshots

image

# Test Coverage

✅

### CODE REVIEW

File: client/app/components/Input/InputError.jsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
+// @flow
+import React from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import {
+  faAsterisk,
```
Comment (julianguyen) [inline client/app/components/Input/InputError.jsx]: This line should be removed cause it's not being used.

File: client/app/components/Input/InputError.jsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  faQuestion,
```
Comment (julianguyen) [inline client/app/components/Input/InputError.jsx]: This line should be removed cause it's not being used.

File: client/app/components/Input/InputError.jsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  faExclamation,
+} from '@fortawesome/free-solid-svg-icons';
+import { Tooltip } from '../Tooltip';
```
Comment (julianguyen) [inline client/app/components/Input/InputError.jsx]: This line should be removed cause it's not being used.

File: client/app/components/Input/InputError.jsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  label: string,
+  required?: boolean,
+  info?: string,
+  error?: boolean,
+};
+
+
+export const InputError = (props: Props) => {
+  const {
+    error, label, required, info,
```
Comment (julianguyen) [inline client/app/components/Input/InputError.jsx]: Label, required, and info should be removed cause it's not being used.

File: client/app/components/Input/InputLabel.jsx [hunk 5]
```diff
@@ -9,6 +9,7 @@ import {
 import { Tooltip } from '../Tooltip';
 import css from './Input.scss';
 import globalCss from '../../styles/_global.scss';
+import { I18n } from '../../libs/i18n';
```
Comment (julianguyen) [inline client/app/components/Input/InputLabel.jsx]: This line should be removed cause it's not being used.

File: client/app/components/Input/InputError.jsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  faQuestion,
+  faExclamation,
+} from '@fortawesome/free-solid-svg-icons';
+import { Tooltip } from '../Tooltip';
+import css from './Input.scss';
+import globalCss from '../../styles/_global.scss';
+import { I18n } from '../../libs/i18n';
+
+export type Props = {
+  label: string,
```
Comment (julianguyen) [inline client/app/components/Input/InputError.jsx]: Let's remove label, required, and info from props.

File: client/app/components/Input/index.jsx [hunk 7]
```diff
@@ -236,7 +237,27 @@ export class Input extends React.Component<Props, State> {
...
 
+  displayError = () => {
+    const {
+      label, info, required, type,
+    } = this.props;
+    const { error } = this.state;
+    if (REQUIRES_LABEL.includes(type) && label) {
+      return (
+        <InputError
+          label={label}
```
Comment (julianguyen) [inline client/app/components/Input/index.jsx]: Let's remove label, required, and info from props.

File: client/app/components/Input/InputError.jsx [hunk 8]
```diff
@@ -0,0 +1,39 @@
+// @flow
```
Comment (julianguyen) [inline client/app/components/Input/InputError.jsx]: Let's write a test for this! Check out the `__tests__` folder in Input for examples. Let me know if you want to pair on this :)

File: client/app/components/Input/InputError.jsx [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+  error?: boolean,
+};
+
+
+export const InputError = (props: Props) => {
+  const {
+    error, label, required, info,
+  } = props;
+  return (
+    <div className={`${globalCss.gridRowSpaceBetween} ${css.label}`}>
```
Comment (julianguyen) [inline client/app/components/Input/InputError.jsx]: Let's remove this

File: client/app/components/Input/InputError.jsx [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+    <div className={css.labelInfo}>
+    {error ? (
+        <div className="labelError">
```
[the remark this answers is not in the document]
Comment (Mera-Gangapersaud) [reply client/app/components/Input/InputError.jsx]: After removing the div on line 27 and adding the smallMarginTop class the font of the label error becomes black and looks like this:

screenshot from 2018-11-22 09-50-45


I tried adding 
```
<div className={`labelError ${globalCss.smallMarginTop} ${css.label}`}>
```
 instead and it looks like this:
 
screenshot from 2018-11-22 09-45-40

Is the pink or black preferable?

**Comment by Mera-Gangapersaud** [comment]
@julianguyen I accidentally deleted my branch when trying to push some changes which closed this PR :confounded:  
Is it possible for you to reopen it?

**Comment by julianguyen** [comment]
@Mera-Gangapersaud Oh no! Hmm, looks like I can't from my end
image

Do you still have your branch your local environment?

**Comment by Mera-Gangapersaud** [comment]
I do

**Comment by julianguyen** [comment]
Yeah @Mera-Gangapersaud yeah, feel free to open a new PR!

File: client/app/components/Input/InputError.jsx (hunk 10 above)
[the remark this answers is not in the document]
Comment (julianguyen) [reply client/app/components/Input/InputError.jsx]: We want it to be the original pinkish colour!

**Pull request closed without merging**